src/main/catalog-pusher.ts

```diff
@@ -0,0 +1,32 @@
+import { autorun, toJS } from "mobx";
+import { broadcastMessage, subscribeToBroadcast, unsubscribeFromBroadcast } from "../common/ipc";
+import { CatalogEntityRegistry} from "../common/catalog-entity-registry";
+import "../common/catalog-entities/kubernetes-cluster";
+
+export class CatalogPusher {
+  static init(catalog: CatalogEntityRegistry) {
+    new CatalogPusher(catalog).init();
+  }
+
+  private constructor(private catalog: CatalogEntityRegistry) {}
+
+  init() {
+    const disposers: { (): void; }[] = [];
+
+    disposers.push(autorun(() => {
+      this.broadcast();
+    }));
+
+    const listener = subscribeToBroadcast("catalog:broadcast", () => {
+      this.broadcast();
+    });
+
+    disposers.push(() => unsubscribeFromBroadcast("catalog:broadcast", listener));
+
+    return disposers;
+  }
+
+  broadcast() {
+    broadcastMessage("catalog:items", toJS(this.catalog.items, { recurseEverything: true }));
```

Comment:
We should remember to document that catalogue entities *must* be IPC serializable.

Reply:
Maybe we could add `toJS` (or similar) method to `CatalogEntity` interface?

Reply:
`toJS` merely removes observables. If you look at the serializable documentation things like `functions` and class hierarchy are not preserved across IPC serialization.